kubernetes-itests/src/test/java/io/fabric8/kubernetes/TypedCustomResourceIT.java

```diff
@@ -59,14 +62,53 @@
 
   private String currentNamespace;
 
+  private static final CustomResourceDefinition petCrd = new CustomResourceDefinitionBuilder()
+    .withNewMetadata().withName("pets.testing.fabric8.io").endMetadata()
+    .withNewSpec()
+    .withGroup("testing.fabric8.io")
+    .addNewVersion()
+    .withName("v1alpha1")
+    .withServed(true)
+    .withStorage(true)
+    .withNewSubresources()
+    .withNewStatus().endStatus()
+    .endSubresources()
+    .withNewSchema()
+    .withNewOpenAPIV3Schema()
+    .withType("object")
+    .addToProperties(Collections.singletonMap("spec", new JSONSchemaPropsBuilder()
+      .withType("object")
+      .withProperties(Collections.singletonMap("type", new JSONSchemaPropsBuilder()
+        .withType("string")
+        .build()))
+      .build()))
+    .addToProperties(Collections.singletonMap("status", new JSONSchemaPropsBuilder()
+      .withType("object")
+      .withProperties(Collections.singletonMap("currentStatus", new JSONSchemaPropsBuilder()
+        .withType("string")
+        .build()))
+      .build()))
+    .endOpenAPIV3Schema()
+    .endSchema()
+    .endVersion()
+    .withScope("Namespaced")
+    .withNewNames()
+    .withPlural("pets")
+    .withSingular("pet")
+    .withKind("Pet")
+    .withShortNames("pt")
+    .endNames()
+    .endSpec()
+    .build();
```

Comment:
Most of the boilerplate can be skipped, see https://github.com/fabric8io/kubernetes-client/blob/22e2e6fe942b9a740812ba2d45968a31e0e045da/kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/CustomResourceV1Example.java as reference

Reply:
Umm, strange. Why is this method part of `CustomResourceDefinitionContext`? It doesn't seem to be related to `CustomResourceDefinitionContext`

Reply:
It's not, this was discussed in scope #266.
IIRC we all agreed these methods had to live somewhere for now.

Reply:
Maybe you linked the wrong issue... When I looked at your example, it seemed a bit weird. I thought maybe `CustomResource` is the right place for that method

Reply:
Wrong issue indeed. I don't see `CustomResource` a good fit either for that, I think it might even be worse since it will add even more confusion to the CustomResource vs. CustomResourceDefinition situation.
The helper is used to create a CustomResourceDefinition from a CustomResource pojo. It might make sense to make it part of CustomResourceDefinition class, but of course that's a generated model, and there are (as of now) 2 of them.
IMHO it makes sense to have these in an utils class. I'm positive `CustomResourceDefinitionContext` isn't a good place either.